robot_smach_states/src/robot_smach_states/world_model.py

```diff
@@ -243,15 +243,19 @@ def __init__(self, robot, segmented_entity_ids_designator, entity_designator, vo
         self.entity_des = entity_designator
 
         ds.check_type(volume_designator, str)
-        # Can be either designator or concrete value, but the resolve function takes care of either case
-        self.volume_designator = volume_designator
+        if isinstance(volume_designator, str):
+            self.volume_designator = ds.VariableDesignator(volume_designator)
+        elif isinstance(volume_designator, ds.Designator):
+            self.volume_designator = volume_designator
+        else:
+            raise RuntimeError("This shouldn't happen. Wrong types should have raised an exception earlier")
```

Comment:
If we want to enforce that only designators are used for this function we should not add support for string inputs. In that case even the check_type above this should be replaced with
assert(volume_designator.resolve_type is <str>)